Avoid unnecessary query to wp_options on every request

rodrigoprimo opened the pull request:
This PR changes the value of the option 'woocommerce_queue_flush_rewrite_rules' to 'no' instead of deleting it when rewrite rules are flushed. This way this option is autoloaded and the `get_option()` call below doesn't trigger a select query to the database on every request.

https://github.com/woocommerce/woocommerce/blob/61f037060b74ad9be8fb2d63e5ea3050fd95fb1e/includes/class-wc-post-types.php#L514

It also changes the value of the option 'woocommerce_queue_flush_rewrite_rules' to 'yes' or 'no' instead of 'true' or 'false' because the former seems to be the standard used by WC core and because the string 'false' can induce bugs as it is truthy.

The problem fixed here was introduced in WC 3.3.0, but it is not really a bug so I'm not sure if this PR should be merged included in 3.3.2 or 3.4.0 (cc @mikejolley).

rodrigoprimo committed:
Use 'yes' instead of 'true' as the value of the option woocommerce_queue_flush_rewrite_rules

WC seems to use 'yes' and 'no' instead of 'true' and 'false' as the values of boolean options, so this commit changes the value of the option woocommerce_queue_flush_rewrite_rules to follow this pattern. Also, the string 'false' can induce bugs as it is truthy.

rodrigoprimo committed:
Avoid unneeded query to wp_options on every request

This commit changes the value of the option 'woocommerce_queue_flush_rewrite_rules' to 'no' instead of deleting it when rewrite rules are flushed. This way this option is autoloaded and the `get_option()` call below doesn't trigger a select query to the database on every request.

https://github.com/woocommerce/woocommerce/blob/61f037060b74ad9be8fb2d63e5ea3050fd95fb1e/includes/class-wc-post-types.php#L514

mikejolley commented:
@rodrigoprimo This changes the delete option to an update option. Because of this, the `add_option` calls will do nothing because the option will already exist.

rodrigoprimo committed:
Update option instead of adding it

Doing this since on commit 57ae11c4c6e7ab2b9b747e8326b48504fcb99fd7, we stopped deleting the option and thus adding it won't do anything as it will be already present.

rodrigoprimo commented:
Good catch Mike, thanks. On commit b4f6dc4, I replaced the calls to `add_option()` with calls to `update_option()`.

rodrigoprimo committed:
Merge branch 'master' into update/avoid-extra-unneeded-query
